src/RequestHeaders.php

```diff
@@ -39,11 +41,14 @@ public function get(string $key): array
      */
     public function add(string $key, string $value): void
     {
-        $lowercaseKey = strtolower($key);
-        if (array_key_exists($lowercaseKey, $this->headers)) {
-            $this->headers[$lowercaseKey][$value] = true;
+        $key = $this->normalize($key);
+        $value = $this->normalize($value);
```

Comment:
Normalized the value to prevent adding the same value multiple times with different casing

Reply:
Header values are case sensitive.